### DISCUSSION TITLE
Programmatically manage projects via Vercel API / SDK

### DISCUSSION THREAD

**Discussion by gyenabubakar** [vercel/community#4583] [Feature Question]
### Question

My team is thinking about about building our own headless-CMS-like system and I was wondering if there's a Vercel API or SDK that supports the following:

1. Adding a new project or "site" connecting it to a private (preferred) or public repo. Updates to the main branch of the repo should trigger deployment on Vercel.

2. Setting custom domain names.

**Comment by amyegan** [comment]
Vercel REST API has endpoints available to [create a new project](https://vercel.com/docs/rest-api/endpoints#create-a-new-project) and [add a domain to a project](https://vercel.com/docs/rest-api/endpoints#add-a-domain-to-a-project) 😄 

https://vercel.com/docs/rest-api
